src/cascadia/WindowsTerminal/WindowEmperor.cpp

```diff
@@ -18,600 +30,1024 @@ using namespace ::Microsoft::Console;
+    // * On Windows 10 the CoreWindow will show up as a visible window on the
+    //   taskbar due to a WinUI bug, and this will hide it.
+    // * When we DestroyWindow() the island it will destroy the CoreWindow,
+    //   and it's not possible to recreate it. That's also a WinUI bug.
+    if (const auto coreWindow = winrt::Windows::UI::Core::CoreWindow::GetForCurrentThread())
+    {
+        if (const auto interop = coreWindow.try_as<ICoreWindowInterop>())
+        {
+            HWND coreHandle = nullptr;
+            if (SUCCEEDED(interop->get_WindowHandle(&coreHandle)) && coreHandle)
+            {
+                SetParent(coreHandle, _window.get());
+            }
+        }
+    }
 
-                // On Windows 11, we DONT want to refrigerate the window. There,
-                // we can just close it like normal. Break out of the loop, so
-                // we don't try to put this window in the fridge.
-                if (Utils::IsWindows11())
-                {
-                    decrementWindowCount.reset();
-                    break;
-                }
-                else
-                {
-                    window->Refrigerate();
-                    decrementWindowCount.reset();
+    // Main message loop. It pumps all windows.
+    bool loggedInteraction = false;
+    MSG msg{};
+    while (GetMessageW(&msg, nullptr, 0, 0))
+    {
+        if (!loggedInteraction && (msg.message == WM_KEYDOWN || msg.message == WM_SYSKEYDOWN))
+        {
+            TraceLoggingWrite(
+                g_hWindowsTerminalProvider,
+                "SessionBecameInteractive",
+                TraceLoggingLevel(WINEVENT_LEVEL_VERBOSE),
+                TelemetryPrivacyDataTag(PDT_ProductAndServiceUsage));
+            loggedInteraction = true;
+        }
 
-                    if (auto self{ weakThis.lock() })
-                    {
-                        auto fridge{ self->_oldThreads.lock() };
-                        fridge->push_back(window);
-                    }
-                }
+        // GH#638 (Pressing F7 brings up both the history AND a caret browsing message)
+        // The Xaml input stack doesn't allow an application to suppress the "caret browsing"
+        // dialog experience triggered when you press F7. Official recommendation from the Xaml
+        // team is to catch F7 before we hand it off.
+        // AppLogic contains an ad-hoc implementation of event bubbling for a runtime classes
+        // implementing a custom IF7Listener interface.
+        // If the recipient of IF7Listener::OnF7Pressed suggests that the F7 press has, in fact,
+        // been handled we can discard the message before we even translate it.
+        if ((msg.message == WM_KEYDOWN || msg.message == WM_SYSKEYDOWN) && msg.wParam == VK_F7)
+        {
+            if (const auto w = _mostRecentWindow(); w && w->OnDirectKeyEvent(VK_F7, LOBYTE(HIWORD(msg.lParam)), true))

```

Comment:
The HWND we're stealing the input from is the CoreInput window - `Windows.UI.Input.InputSite.WindowClass`. Because of that we don't know what HWND to correlate the message with.